### PR TITLE
Fix 729

### LINKED ISSUE #729: `ChainId` should serialize itself without using `tendermint::chain::Id`
Currently, the `serde::Serialization` is achieved by [first converting](https://github.com/cosmos/ibc-rs/blob/62575ceb2a233cc78bbf7324e826eeb95ccb91a1/crates/ibc/src/core/ics24_host/identifier.rs#L46) to `tendermint::chain::Id`, and then serializing from there. We should instead implement `Serialize` and `Deserialize` manually by first converting to a `String`, then serializing that `String`, and vice-versa, as we do with the `tendermint::chain::Id` type.

This is part of the effort to completely remove the dependency on Tendermint from core IBC.

### PR DESCRIPTION
**Pull request by DaviRain-Su**
Closes: #729 

## Description

______

### PR author checklist:
- [ ] Added changelog entry, using [`unclog`](https://github.com/informalsystems/unclog).
- [ ] Added tests.
- [x] Linked to GitHub issue.
- [ ] Updated code comments and documentation (e.g., `docs/`).
- [ ] Tagged *one* reviewer who will be the one responsible for shepherding this PR.

### Reviewer checklist:

- [ ] Reviewed `Files changed` in the GitHub PR explorer.
- [ ] Manually tested (in case integration/unit/mock tests are absent).

### CODE REVIEW

File: crates/ibc/src/core/ics02_client/handler/update_client.rs [hunk 1]
```diff
@@ -329,7 +329,9 @@ mod tests {
             let client_state = {
                 #[allow(deprecated)]
                 let raw_client_state = RawTmClientState {
-                    chain_id: ChainId::from(tm_block.header().chain_id.clone()).to_string(),
+                    chain_id: ChainId::from_str(tm_block.header().chain_id.to_string().as_str())
```
Comment (Farhad-Shabani) [inline crates/ibc/src/core/ics02_client/handler/update_client.rs]: Here we can go away with: `ChainId::from(tm_block.header().chain_id.to_string()).to_string()`

File: crates/ibc/src/core/ics24_host/identifier.rs [hunk 2]
```diff
@@ -165,15 +161,9 @@ impl Display for ChainId {
...
-        tendermint::chain::Id::from_str(id.as_str()).unwrap()
-    }
-}
-
-impl From<tendermint::chain::Id> for ChainId {
-    fn from(id: tendermint::chain::Id) -> Self {
-        ChainId::from(id.to_string())
+        id.to_string()
     }
 }
```
Comment (Farhad-Shabani) [inline crates/ibc/src/core/ics24_host/identifier.rs]: This conversation looks redundant imo. `to_string()` is straightforward enough if anyone needs it.

File: crates/ibc/src/clients/ics07_tendermint/client_state.rs [hunk 3]
```diff
@@ -1127,7 +1127,7 @@ pub mod test_util {
 
         pub fn new_dummy_from_header(tm_header: Header) -> ClientState {
             ClientState::new(
-                tm_header.chain_id.clone().into(),
+                tm_header.chain_id.to_string().try_into().unwrap(),
```
Comment (Farhad-Shabani) [inline crates/ibc/src/clients/ics07_tendermint/client_state.rs]: Likewise, we can use `.to_string().into()` instead

**Review by Farhad-Shabani — COMMENTED** [review]
Thanks for taking care of this. Left a few comments.
Plus, it needs a changelog entry.
Would you also include a changelog for #731 along with this PR?

File: crates/ibc/src/core/ics24_host/identifier.rs (hunk 2 above)
Comment (DaviRain-Su) [reply crates/ibc/src/core/ics24_host/identifier.rs]: have removed

File: crates/ibc/src/core/ics02_client/handler/update_client.rs (hunk 1 above)
Comment (DaviRain-Su) [reply crates/ibc/src/core/ics02_client/handler/update_client.rs]: have fixed!

File: crates/ibc/src/clients/ics07_tendermint/client_state.rs (hunk 3 above)
Comment (DaviRain-Su) [reply crates/ibc/src/clients/ics07_tendermint/client_state.rs]: have fixed!

**Review by Farhad-Shabani — APPROVED** [review]
🙏🏻

**Pull request merged**